docs/architecture/deep-dives/passkeys/operations.md

```diff
@@ -0,0 +1,148 @@
+---
+sidebar_position: 3
+---
+
+# Operations
+
+This document will provide an overview of two main operations performed through the WebAuthn API:
+registering a new passkey (registration) and signing in (authentication).
+
+## Registering a Passkey
+
+Registration requests are invoked using the `navigator.credentials.create()` method.
+
+1. **Client-Side Server Request:** The application (usually a JavaScript application running in a
+   web browser) initiates the registration process by making a request to the server to create a new
+   FIDO2 credential.
+
+2. **Server-Side Preparation:** The server prepares for the registration process by generating an
+   options object, usually with the help of a code library (e.g. `fido2-net-lib`). This object
+   includes the following parameters:
+
+   - **attestation**: The attestation statement format to use during the registration process. This
+     can be set to `none` if the server does not require attestation.
+   - **authenticatorSelection**: An object that specifies the authenticator attachment modality and
+     the user verification requirement.
+   - **challenge**: A random value generated by the server that will be used to verify the
+     authenticity of the subsequent response from the client.
+   - **excludeCredentials**: A list of existing credentials that the server should exclude from the
+     registration process. This is useful for preventing duplicate registrations, by aborting the
+     registration if the authenticator finds an existing credential that matches one of the
+     credentials in this list.
+   - **pubKeyCredParams**: A list of public key parameters that the server is willing to accept for
+     the new credential. This includes the cryptographic algorithm and the key type.
+   - **rp**: An object that contains information about the Relying Party (RP). This includes the RPs
+     name, ID, and icon.
+   - **timeout**: The time, in milliseconds, that the client is willing to wait for the
+     authenticator to complete the registration process.
+   - **user**: An object that contains information about the user. This includes the user's ID,
+     name, display name, and icon.
+   - **extensions**: An object that contains any additional parameters that the server wants to
+     include in the registration process. An example of this is the `PRF` extension, which enables
```

Comment:
❓ Is the convention to capitalize "PRF" or lowercase (`PRF` vs `prf`)?  As an initialism it should be capitalized, but referencing it in code in the JSON data it's always `prf`.

Reply:
Good point! If we monospace it then it should be `prf` to mirror the codebase. It also shouldn't be monospaced here, removed the formatting.

Regarding `PRF` vs `prf`:
The codebase uses strict camelCase, i.e. we (mostly) ignore things like initialisms and only uppercase when it's a new word.  E.g. we write `cipherId` and not `cipherID`